src/formatDate.js

```diff
@@ -1,14 +1,34 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      receivedDate['YY'] = dateArr[i].slice(2);
+    }
+
+    if (fromFormat[i] === 'YY') {
+      receivedDate['YYYY'] = (dateArr[i] < 30 ? 20 : 19) + dateArr[i];
+    }
+
+    receivedDate[fromFormat[i]] = dateArr[i];
+  }
+
+  for (const elem of toFormat.slice(0, -1)) {
+    changedDate.push(receivedDate[elem]);
```

Comment:
There is a potential issue here if the `toFormat` array contains 'YYYY' or 'YY' and the corresponding key does not exist in the `receivedDate` object due to the earlier mentioned issues. Make sure that all keys are correctly assigned in the `receivedDate` before this loop.